### PR TITLE
9281 chainHub exo

### PR DESCRIPTION
**Pull request by turadg**
refs: #9281 

## Description

AsyncFlow requires that everything passing the membrane is durable. This makes the ChainHub durable by making it an Exo.

### Security Considerations
none

### Scaling Considerations

It's a singleton, because there will only be one per vat.

### Documentation Considerations
none

### Testing Considerations
Existing coverage

### Upgrade Considerations
Not yet deployed.

Now that it's an Exo, we have to consider its state an upgradability. We expect to make a new one on each vat invocation. The only state it has is cache of AgoricNames and data supplied by the contract.

### CODE REVIEW

File: packages/orchestration/src/exos/chain-hub.js [hunk 1]
```diff
@@ -54,7 +54,38 @@ export const connectionKey = (chainId1, chainId2) => {
   return [chainId1, chainId2].sort().join(CHAIN_ID_SEPARATOR);
 };
 
+const ChainIdArgShape = M.or(
```
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/chain-hub.js:57]: consider:
```suggestion
/** accepts `chainId` or `ChainAddress` object */
const ChainIdArgShape = M.or(
```

**Review by 0xpatrickdev — APPROVED** [review]

File: packages/orchestration/src/exos/chain-hub.js (hunk 1 above)
Comment (turadg) [reply packages/orchestration/src/exos/chain-hub.js:57]: good idea. I'll make a note to get that into a future PR

File: packages/orchestration/src/exos/chain-hub.js [hunk 2]
```diff
@@ -54,7 +54,38 @@ export const connectionKey = (chainId1, chainId2) => {
...
+
+const ChainHubI = M.interface('ChainHub', {
+  registerChain: M.call(M.string(), CosmosChainInfoShape).returns(),
+  getChainInfo: M.callWhen(M.string()).returns(CosmosChainInfoShape),
+  registerConnection: M.callWhen(
+    M.string(),
+    M.string(),
+    IBCConnectionInfoShape,
+  ).returns(),
+  getConnectionInfo: M.callWhen(ChainIdArgShape, ChainIdArgShape).returns(
```
Comment (dckc) [inline packages/orchestration/src/exos/chain-hub.js:76]: `callWhen` is OK in asyncFlow after all?

I can imagine waiting for tests that show that it's not, but in that case, why not wait for tests that show that chainHub has to be an exo in the 1st place?

Comment (turadg) [reply packages/orchestration/src/exos/chain-hub.js:76]: Nope, I learned last night it's not.

We did hit a test failure with the object itself not being durable but only in another branch that had a higher fidelity asyncFlow

**Pull request merged**